rancher2/0_provider_upgrade_test.go

```diff
@@ -427,15 +427,21 @@ func testAccRancher2UpgradeRancher() resource.TestCheckFunc {
 
 func testAccRancher2UpgradeVars() resource.TestCheckFunc {
 	return func(s *terraform.State) error {
-		for _, rs := range s.RootModule().Resources {
+		for k, rs := range s.RootModule().Resources {
 			if rs.Type != "rancher2_bootstrap" {
 				continue
 			}
-
 			token := rs.Primary.Attributes["token"]
-			os.Setenv("RANCHER_TOKEN_KEY", token)
+			rancherTokenKey := os.Getenv("RANCHER_TOKEN_KEY")
+			rancherAdminPass := os.Getenv("RANCHER_ADMIN_PASS")
+			if err := os.Setenv("RANCHER_TOKEN_KEY", token); err != nil {
+				fmt.Printf("Failed to update RANCHER_TOKEN_KEY on ressource %s with err %s", k, err.Error())
+			}
 			currentPassword := rs.Primary.Attributes["current_password"]
-			os.Setenv("RANCHER_ADMIN_PASS", currentPassword)
+			if err := os.Setenv("RANCHER_ADMIN_PASS", currentPassword); err != nil {
+				fmt.Printf("Failed to update RANCHER_ADMIN_PASS on ressource %s with err %s", k, err.Error())
+			}
+			fmt.Printf("Changing env on ressource %s: RANCHER_TOKEN_KEY: %s -> %s; RANCHER_ADMIN_PASS: %s -> %s", k, rancherTokenKey, token, rancherAdminPass, currentPassword)
```

Comment:
Yes, that is exactly what it is doing. 

I added some comments to that function, mainly I was having a flaky test (more than 90% of the cases) where I was getting an unauthorized and rancher was letting some dangling resources. 

I added those prints to understand what was happening.   After adding that print I didn't got that error anymore. So I left them. 

Do you think that it is Ok to keep the token get / print ?

Reply:
As it stands I don't feel confident merging this if the issue is no longer there because of some print statements. If there is a race condition, we need to fix it.

Reply:
I'll take a deeper look.

I imagined that could be a race cause the print statements fixed it. 

Those errors just happens on tests and that function is a test one.

Reply:
Thanks for asking this. 

At the end everything that I tried  wasn't 100% Then I found out reading the Struct that is passed to tests that they don't support "Aliases" on the test framework and that was causing the flaky test.,